### PR TITLE
Update security

### PR DESCRIPTION
**Pull request by OH72**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -109,7 +109,7 @@
 		<dependency>
 			<groupId>org.projectlombok</groupId>
 			<artifactId>lombok</artifactId>
-			<version>1.18.18</version>
+			<version>1.18.20</version>
```
Comment (OH72) [inline pom.xml:112]: failed to build with previous version

File: src/main/java/com/softserve/teachua/config/SecurityConfig.java [hunk 2]
```diff
@@ -110,75 +110,94 @@ protected void configure(HttpSecurity http) throws Exception {
                         "/centers",
                         "/center/*",
                         "/service").permitAll()
+                .antMatchers("/api/**").permitAll()
```
Comment (OH72) [inline src/main/java/com/softserve/teachua/config/SecurityConfig.java:113]: I commented all lines that are relative to api/... and added security on each method separately.

File: src/main/java/com/softserve/teachua/controller/AboutUsItemController.java [hunk 3]
```diff
@@ -35,6 +36,7 @@ public AboutUsItemResponse getAboutUsItems(
         return aboutUsItemService.getAboutUsItemResponseById(id);
     }
 
+    @PreAuthorize("hasRole(T(com.softserve.teachua.constants.RoleData).ADMIN.getDBRoleName())")
```
Comment (OH72) [inline src/main/java/com/softserve/teachua/controller/AboutUsItemController.java:39]: It means that only ADMIN can execute this method.

File: src/main/java/com/softserve/teachua/controller/CenterController.java [hunk 4]
```diff
@@ -63,6 +64,8 @@ public CenterResponse getCenter(@PathVariable Long id) {
      *
      * @return new {@code SuccessCreatedCenter}.
      */
+    @PreAuthorize("hasAnyRole(T(com.softserve.teachua.constants.RoleData).ADMIN.getDBRoleName(), " +
+            "T(com.softserve.teachua.constants.RoleData).MANAGER.getDBRoleName())")
```
Comment (OH72) [inline src/main/java/com/softserve/teachua/controller/CenterController.java:68]: It means that only ADMIN and MANAGER can execute this method.

File: src/main/java/com/softserve/teachua/controller/ComplaintController.java [hunk 5]
```diff
@@ -63,6 +64,7 @@ public ComplaintResponse getComplaintById(@PathVariable Long id) {
      * @param complaintProfile - object of DTO class
      * @return new {@link SuccessCreatedComplaint}
      */
+    @PreAuthorize("isAuthenticated()")
```
Comment (OH72) [inline src/main/java/com/softserve/teachua/controller/ComplaintController.java:67]: It means that only authenticated user can execute this method.

**Review by RomanKlymus — APPROVED** [review]

**Pull request merged**